player/lua/ytdl_hook.lua

```diff
@@ -690,8 +692,11 @@ function run_ytdl_hook(url)
     -- check for youtube-dl in mpv's config dir
     if not (ytdl.searched) then
         local exesuf = (package.config:sub(1,1) == '\\') and '.exe' or ''
-        local ytdl_mcd = mp.find_config_file("youtube-dl" .. exesuf)
-        if not (ytdl_mcd == nil) then
+        local ytdl_mcd = mp.find_config_file(o.ytdl_path .. exesuf)
+        if ytdl_mcd == null then
+            msg.verbose("No youtube-dl found with path "..o.ytdl_path..exesuf.." in config directories")
+            ytdl.path = o.ytdl_path
```

Comment:
1) `nil` not `null`
2) Missing exesuf on this line

But what I actually want to note here is: I think it'd be more useful to use ytdl_path as-is if it's specified by the user(!) instead of calling doing the find config file thing

Reply:
>     2\. Missing exesuf on this line
> But what I actually want to note here is: I think it'd be more useful to use ytdl_path as-is if specified by the user(!) instead of doing the find config file thing

That's to support placing the executable in the config dirs.

>     1. `nil` not `null`

Too much javascript.

Reply:
Btw, @sfan5, I'd personally argue that using something like `--script-opts=ytdl_hook-ytdl_path=ytdl` could still work, without you needing to explicitely pass the absolute path.

Reply:
Yeah that sounds fine.